Add guards to unguarded traces

MrTrader8 opened the pull request:
In the code there are places where unguarded tracing occurs causing
an output into a log file when no trace flags are specified.
This commit adds guards to stop the unguarded tracing.

Signed-off-by: Andrew Gao <andrewgao98@gmail.com>

andrewcraik commented on runtime/compiler/optimizer/DataAccessAccelerator.hpp:
```diff
@@ -243,14 +243,15 @@ class TR_DataAccessAccelerator : public TR::Optimization
-      if (status)
-         traceMsg(comp(), "DataAccessAccelerator: Intrinsics on node %p : SUCCESS\n", node);
-      else
+      if (trace()){
```
please match code formatting in surrounding code for the {}s

andrewcraik commented on runtime/compiler/optimizer/SPMDParallelizer.cpp:
```diff
@@ -3180,12 +3180,14 @@ bool TR_SPMDKernelParallelizer::processGPULoop(TR_RegionStructure *loop, TR_SPMD
    if (lastNode->getOpCode().isGoto())
       {
+      if (trace())
       traceMsg(comp(), "Changing destination of goto at the end of block_%d\n", loopInvariantBlock->getNumber());
```
please indent traceMsg

andrewcraik commented:
The chatter from SPMDPreCheck is pretty important because it will prevent a number of loop transformations leading up to SPMDParallization. I am of the opinion these should probably remain chatting - there are not many messages since it is at most one per loop.

I'm not sure I am a fan of passing the whole CISCTransformer around in IdiomRecognition to test if tracing is enabled. Wouldn't it be cleaner to have an argument holding the tracing state to avoid extra dependencies being added to the transformer?

MrTrader8 commented:
@andrewcraik I have made the changes to the formatting of the code, as for SPMDPreCheck, guards were added since they were outputting quite a bit of lines(over 50+), however I can remove the guards if needed. In the case for CISCTransformer being passed around, would passing in disptrace as an additional boolean parameter to keep track of tracing be a better option?

andrewcraik commented:
I personally would prefer the bool because it would stop additional dependencies onto the transformer being introduced. For SPMDPreCheck I still think the output is useful in a traceFull scenario because it tells you what loops we gave up on and won't look at further...

MrTrader8 commented:
@andrewcraik I have changed the functions to pass in a bool instead of CISCTransformer to keep track of the tracing, additionally for the SPMDPreCheck I have changed the tracing guards to print in a TraceFull scenario.

andrewcraik commented on runtime/compiler/optimizer/IdiomTransformations.cpp:
```diff
@@ -1086,8 +1096,9 @@ findIndVarLoads(TR::Node *node, TR::Node *indVarStoreNode, bool &storeFound,
 
 static int32_t
-checkForPostIncrement(TR::Compilation *comp, TR::Block *loopHeader, TR::Node *loopCmpNode, TR::Symbol *ivSym)
+checkForPostIncrement(TR_CISCTransformer *trans, TR::Compilation *comp, TR::Block *loopHeader, TR::Node *loopCmpNode, TR::Symbol *ivSym)
```
Was this one going to be updated with a bool as well?

andrewcraik commented on runtime/compiler/optimizer/SPMDPreCheck.cpp:
```diff
@@ -29,11 +29,13 @@
 bool SPMDPreCheck::isSPMDCandidate(TR::Compilation *comp, TR_RegionStructure *loop)
    {
- 
+   bool trace = comp->getOption(TR_TraceAll);
```
Shoudl this be `comp->getOption(TR_TraceALL) || trace()` to make sure the trace turns on if only SPMDPreCheck is being traced?

gita-omr commented:
@MrTrader8 do you recall what prevented us from completing this PR?

MrTrader8 commented:
@gita-omr I believe this PR was just not merged, if i recall correctly the omr side of the PR was merged already.

gita-omr commented:
Jenkins test sanity all jdk8,jdk11

gita-omr commented:
@MrTrader8 most likely you will need to rebase your changes. Let's try to merge it soon.

gita-omr commented:
This work became a part of #9586, so  closing.
